waitress/app/views.py

```diff
@@ -13,3 +18,33 @@
     public=True,
     permission_classes=(AllowAny,),
 )
+
+
+def login_handler(request):
+    if request.user.is_authenticated:
+        return redirect("dashboard")
+
+    context = dict(login_form=LoginForm)
+    if request.method == "POST":
+        error_message = "Username or Password incorrect."
+        username = request.POST.get("username")
+        password = request.POST.get("password")
+        form = LoginForm(request.POST)
+
+        if not form.is_valid():
```

Comment:
Actually, You can use class based views that inherits from `View` and use def `post` as you see fit.
This has a lot of problems, 
1. nexted condition, would be hard for testing and tracking
2. too many conditions for testing if it's post or get

Reply:
Fixed.